### PR TITLE
chore: add kotlin code splitter

### PR DESCRIPTION
**Pull request by fynnfluegge**
- **Description:** Adds Kotlin language to `TextSplitter`

### CODE REVIEW

**Comment by fynnfluegge** [comment]
I can also add an example to [code_splitter.mdx](https://python.langchain.com/docs/modules/data_connection/document_transformers/text_splitters/code_splitter) but there is not even a java example. Question is, add example for java or kotlin or both?

File: libs/langchain/tests/unit_tests/test_text_splitter.py [hunk 1]
```diff
@@ -525,6 +525,39 @@ def test_java_code_splitter() -> None:
...
+    companion object {
+        @JvmStatic
+        fun main(args: Array<String>) {
+            println("Hello, World!")
+        }
+    }
+}
+    """
+    chunks = splitter.split_text(code)
+    print(chunks)
```
Comment (eyurtsev) [inline libs/langchain/tests/unit_tests/test_text_splitter.py:543]:
```suggestion
```

**Comment by eyurtsev** [comment]
@fynnfluegge  i don't feel strongly about the documentation given that it's all parameterized by an enum, so easy to discover what's available. If you want feel free to contribute both or either kotlin / java examples, but not required for merge.

File: libs/langchain/langchain/text_splitter.py [hunk 2]
```diff
@@ -762,6 +763,32 @@ def get_separators_for_language(language: Language) -> List[str]:
                 " ",
                 "",
             ]
+        elif language == Language.KOTLIN:
+            return [
```
Comment (eyurtsev) [inline libs/langchain/langchain/text_splitter.py:767]: I don't know any kotlin -- should any other keywords be included?

https://kotlinlang.org/docs/keyword-reference.html

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
Can merge pending removal of print and confirmation from author that keyword list is complete / OK

**Comment by bitomukesh** [comment]
@codium-pr

**Comment by bitomukesh** [comment]
@CodiumAI-Agent /describe

**Comment by fynnfluegge** [comment]
> @fynnfluegge  i don't feel strongly about the documentation given that it's all parameterized by an enum, so easy to discover what's available. If you want feel free to contribute both or either kotlin / java examples, but not required for merge.

True, I think also the examples are fine for now 👍

**Pull request merged**